File: extensions/python/functions/custom.py
def resolve_album_tag_custom_id(ctx):
    return str(ctx["source"].get("CUSTOM_ID", ""))

def resolve_album_tag_custom_albumartist(ctx):
    candidates = ["CUSTOM_ALBUMARTIST", "ARTISTARTIST", "ALBUMARTIST"]
    for key in candidates:
        val = ctx["source"].get(key)
        if val: 
            return str(val)
    return "Unknown"

def resolve_album_tag_custom_string(ctx):
    candidates = ["CUSTOM_STRING", "CUSTOMSTRING"]
    for key in candidates:
        val = ctx["source"].get(key)
        if val: 
            return str(val)
    return ""

def resolve_album_tag_old_comment(ctx):
    return str(ctx["source"].get("OLD_COMMENT", ""))

File: config/extensions/javascript/functions/custom.js
export function resolve_album_tag_custom_id(ctx) {
    return String(ctx.source.CUSTOM_ID || "");
}

export function resolve_album_tag_custom_albumartist(ctx) {
    const candidates = ["CUSTOM_ALBUMARTIST", "ARTISTARTIST", "ALBUMARTIST"];
    for (const key of candidates) {
        if (ctx.source[key]) return String(ctx.source[key]);
    }
    return "Unknown";
}

export function resolve_album_tag_custom_string(ctx) {
    const candidates = ["CUSTOM_STRING", "CUSTOMSTRING"];
    for (const key of candidates) {
        if (ctx.source[key]) return String(ctx.source[key]);
    }
    return "";
}

export function resolve_album_tag_old_comment(ctx) {
    return String(ctx.source.OLD_COMMENT || "");
}
